recipes-connectivity/openssh/openssh_%.bbappend

```diff
@@ -1,4 +1,16 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+SAVED_DIR := "${THISDIR}"
+
+do_patch_append() {
+    bb.build.exec_func('do_close_all_ssh_sessions', d)
+}
+
+do_close_all_ssh_sessions () {
+    # Patch sysvinit file to close all open ssh connections on shutdown or reboot
+    #
+    # perform the patch operation on the version of init in the WORKDIR, not the original recipe
+    patch -u ${WORKDIR}/init -i ${SAVED_DIR}/files/close-all-ssh-connections.patch
```

Comment:
Wait a moment... :thinking:  Do we even need a custom step here? If we're just trying to patch the `/init` file in the source root, after it has already been written by the oe-core layer, then I think a regular `.patch` file should work using the normal bitbake `do_patch` mechanism. I may have led you into reimplementing logic that already exists.

What happens if you replace all these `.bbappend` changes with `SRC_URI += "close-all-ssh-connections.patch"` ? Does bitbake just do the correct thing?

Reply:
No you didn't.  The init file is part of the RECIPE, not part of the source, the default bitbake logic for do_patch operations only applies to the SOURCE.  Its a super fine line, but one that I haven't been able to convince bitbake to cross.

Essentially it would require a patch that executes during do_patch and operates on files in the SOURCE/../init file -- and that won't work.  

I really think this is the right way to go, not as straightforward as a simple .patch file, but it gets the job done and allows for future distro upgrades.

Reply:
Oh; I see. The oe-core `init` script is installed directly into the rootfs. I was thinking that it was a `.patch` and that we could just order our `.patch` after theirs.

In that case, I agree that this is the best approach.